cmd/dep/init.go

```diff
@@ -113,39 +119,26 @@ func (cmd *initCommand) Run(ctx *dep.Ctx, args []string) error {
 	sm.UseDefaultSignalHandling()
 	defer sm.Release()
 
-	pd, err := getProjectData(ctx, pkgT, cpr, sm)
-	if err != nil {
-		return err
-	}
-	m := &dep.Manifest{
-		Dependencies: pd.constraints,
-	}
-
-	// Make an initial lock from what knowledge we've collected about the
-	// versions on disk
-	l := &dep.Lock{
-		P: make([]gps.LockedProject, 0, len(pd.ondisk)),
+	var rootAnalyzer rootProjectAnalyzer
+	var analyzer gps.ProjectAnalyzer
+	if cmd.skipTools {
+		rootAnalyzer = newGopathAnalyzer(ctx.Loggers, ctx, pkgT, cpr, sm)
+		analyzer = dep.Analyzer{}
+	} else {
+		rootAnalyzer = compositeAnalyzer{
+			Analyzers: []rootProjectAnalyzer{
+				newGopathAnalyzer(ctx.Loggers, ctx, pkgT, cpr, sm),
+				newImportAnalyzer(ctx.Loggers, sm),
+			}}
```

Comment:
I think we are both trying to say the same thing. 👍 That's what I was trying to say with "using the importers for both the root project _and_ the dependencies".

The solver takes an an initial manifest and lock for the root project, which I am generating using either just the `gopathAnalyzer` (for `-skip-tools`) or the `compositeAnalayzer`. But it also takes an analyzer to be applied to any dependencies it finds. When not skipping importing from other tools, I give the solver the `importAnalyzer` so that the same import logic we applied at the root is also applied to dependencies.

Thank you for asking about this! I forget to add an integration test for this scenario.